src/Action_GIST.cpp

```diff
@@ -1577,176 +1774,213 @@ void Action_GIST::Print() {
-    }
-    double dTSot = Constants::GASK_KCAL*temperature_*((dTSo/nwtt) + Constants::EULER_MASC);
     infofile_->Printf("watcount in vol = %d\n", nwtt);
     infofile_->Printf("watcount in subvol = %d\n", nwts);
     infofile_->Printf("Total referenced translational entropy of the grid:"
-                      " dTStrans = %9.5f kcal/mol, Nf=%d\n", dTStranstot, NFRAME_);
-    infofile_->Printf("Total 6d if all one vox: %9.5f kcal/mol\n", dTSst);
-    infofile_->Printf("Total t if all one vox: %9.5f kcal/mol\n", dTStt);
-    infofile_->Printf("Total o if all one vox: %9.5f kcal/mol\n", dTSot);
+                      " dTStrans = %9.5f kcal/mol, Nf=%d\n", SumDataSet(*dTStrans_) / NFRAME_, NFRAME_);
+    infofile_->Printf("Total 6d if all one vox: %9.5f kcal/mol\n", SumDataSet(*dTSsix_) / nwts);
+    infofile_->Printf("Total t if all one vox: %9.5f kcal/mol\n", SumDataSet(*dTStrans_) / nwts);
+    infofile_->Printf("Total o if all one vox: %9.5f kcal/mol\n", SumDataSet(*dTSorient_) / nwtt);
+  }
+  // free some memory before allocating all those Farrays for the -dens and -norm data.
+  voxel_xyz_.clear();
+  voxel_xyz_.shrink_to_fit();
+  voxel_Q_.clear();
+  voxel_Q_.shrink_to_fit();
+
+  // Remove solute-solvent energy in voxels without solvent (i.e., at the solute)
+  for (size_t i = 0; i < Esw_->Size(); ++i) {
+    if (N_solvent_[i] == 0) {
+      Esw_->SetGrid(i, 0.0);
+    }
   }
+
   // Compute average voxel energy. Allocate these sets even if skipping energy
   // to be consistent with previous output.
-  DataSet_GridFlt& PME_dens = static_cast<DataSet_GridFlt&>( *PME_);
-  DataSet_GridFlt& U_PME_dens = static_cast<DataSet_GridFlt&>( *U_PME_);
-  DataSet_GridFlt& Esw_dens = static_cast<DataSet_GridFlt&>( *Esw_ );
-  DataSet_GridFlt& Eww_dens = static_cast<DataSet_GridFlt&>( *Eww_ );
-  DataSet_GridFlt& neighbor_norm = static_cast<DataSet_GridFlt&>( *neighbor_norm_ );
-  DataSet_GridDbl& qtet = static_cast<DataSet_GridDbl&>( *order_norm_ );
-  Farray Esw_norm( MAX_GRID_PT_, 0.0 );
-  Farray Eww_norm( MAX_GRID_PT_, 0.0 );
-  Farray PME_norm( MAX_GRID_PT_,0.0);
-  Farray neighbor_dens( MAX_GRID_PT_, 0.0 );
+
+  Darray PME_norm, PME_dens, U_PME_dens;
+  if (usePme_ && !skipE_) {
+    CopyArrayToDataSet(E_pme_, *PME_);
+    CopyArrayToDataSet(U_E_pme_, *U_PME_);
+    mprintf("\t Calculating average voxel energies: \n");
+    PME_norm = NormalizeDataSet<double>(*PME_, N_solvent_);
+    PME_dens = DensityWeightDataSet<double>(*PME_);
+    U_PME_dens = DensityWeightDataSet<double>(*U_PME_);
+    infofile_->Printf("Ensemble total water energy on the grid: %9.5f Kcal/mol \n", SumDataSet(*PME_) / NFRAME_);
+    infofile_->Printf("Ensemble total solute energy on the grid: %9.5f Kcal/mol \n", SumDataSet(*U_PME_) / NFRAME_);
+  }
+  if (n_linear_solvents_ > 0) {
+    mprintf("GIST warning: %d almost-linear solvent molecules occurred. Maybe choose other \"rigidatoms\".\n");
```

Comment:
This printf is missing an argument for the `%d` .

Reply:
Thanks, I meant to print n_linear_solvents.